parachain-upgrade/src/lib.rs

```diff
@@ -53,6 +55,9 @@ pub trait Config: frame_system::Config {
 
 	/// Something which can be notified when the validation data is set.
 	type OnValidationData: OnValidationData;
+
+	/// Returns the parachain ID we are running with.
+	type SelfParaId: Get<ParaId>;
```

Comment:
I was adding this with tears on my eyes, but yeah, turns out that we do need to know the parachains ID. 

Now, we still can get away without it right now. Incidentally, we only need it for stuff that is required for sending messages. So we can just return `None` from this and that should either disable or pause sending until the chain will get the correct para id, presumably set by the governance.

However, if we want to remove `dmq_mqc_head` and `hrmp_mqc_heads` from the `PersistentValidationData`, as per https://github.com/paritytech/polkadot/issues/2244, then the parahchain will have to know its para id otherwise should be there, say, a downward message and a parachain must handle at least one if there is any per parablock.

So I see two solutions here:

1. Add para id to the persisted validation data. It was even mentioned in issues under another angle https://github.com/paritytech/cumulus/issues/221
2. Push polkadot to always generate para id for a parachain before submitting the code.